src/kicanvas/elements/kicanvas-embed.ts

```diff
@@ -98,27 +103,68 @@ class KiCanvasEmbedElement extends KCUIElement {
     async #setup_events() {}
 
     async #load_src() {
-        const sources = [];
+        const url_sources: (string | URL)[] = [];
+        const inline_entries: Record<string, string> = {};
+        const counters: Record<string, number> = {};
 
         if (this.src) {
-            sources.push(this.src);
+            url_sources.push(this.src);
         }
 
         for (const src_elm of this.querySelectorAll<KiCanvasSourceElement>(
             "kicanvas-source",
         )) {
             if (src_elm.src) {
-                sources.push(src_elm.src);
+                url_sources.push(src_elm.src);
+                continue;
+            }
+
+            const content = src_elm.textContent?.trim();
+            if (!content) {
+                continue;
             }
+
+            let filename = src_elm.name?.trim();
+            if (!filename || filename.length == 0) {
+                const type = (src_elm.type ?? "schematic").toLowerCase();
```

Comment:
Similarly to the comment above I've changed this to `src_elm.getAttribute('type')` in my local copy to resolve the issue.